python/tskit/tables.py

```diff
@@ -657,6 +657,17 @@ def asdict(self):
             "child": self.child,
         }
 
+    def squash(self):
+        """
+        Sorts, then condenses the table into the smallest possible number of rows.
+        A single edge replaces any set of edges with the same parent and child node
+        such that the union of (left, right] intervals is a single interval.
+        The new edge will have the same parent and child node, a left coordinate
+        equal to the smallest left coordinate in the set, and a right coordinate
+        equal to the largest right coordinate in the set.
+        """
```

Comment:
We should note whether the edges are in canonical order after this operation (must be, right?), linking to the sortedness requirements for edges in the docs.